### PR TITLE
implement in-db depth first search

### PR DESCRIPTION
**Pull request by adraismawur**
This PR replaces the turns-out-is-slow-and-awful in-database breadth-first implementation with a more efficient depth-first search.

This is currently more memory-efficient than an entirely in-memory DFS, but slower by 2-3 orders of magnitude.

### CODE REVIEW

**Comment by adraismawur** [comment]
And I again forgot to check if query mode is still OK. this does not rely on random selection of edges or whatever, so some of the arguments passed to CC generation are different in query mode. still needs to be checked.

File: big_scape/network/network.py [hunk 1]
```diff
@@ -25,12 +23,14 @@
...
 
+# from this module
+from big_scape.network.DBAdjList import DBAdjList
+
 
 def get_connected_components(
     cutoff: float,
     edge_param_id: int,
     bin: bs_comparison.RecordPairGenerator,
-    seed_record: Optional[BGCRecord] = None,
```
Comment (nlouwen) [inline big_scape/network/network.py:33]: Also remove seed_record docstring

File: big_scape/network/network.py [hunk 2]
```diff
@@ -107,178 +106,150 @@ def get_connected_components(
...
     If a seed record is given, the connected component will be generated starting from that record
+    and the function will return JUST the one connected component
 
     Args:
         cutoff (Optional[float], optional): the distance cutoff. Defaults to None.
         edge_param_id (int): the edge parameter id
         bin_label (str): the bin label
         temp_record_table (Table, optional): a temporary table with the records to include in the
         connected component. Defaults to None.
-        seed_record (BGCRecord, optional): a seed record to start the connected component from.
```
Comment (nlouwen) [inline big_scape/network/network.py:144]: keep seed_record docstring

**Review by nlouwen — CHANGES_REQUESTED** [review]
lgtm, two small docstring changes requested

**Review by nlouwen — APPROVED** [review]
lgtm

**Pull request merged**